### ISSUE TITLE
Update documentation to explain how to specify that Pandas data frame date/time columns contain UTC dates, not local timezone dates

### ISSUE THREAD

**Issue by lloyddewit** [rpy2/rpy2#878]
Thank you for the excellent rpy2 package. We are using rpy2 to analyse climatic data from international weather stations.

We have [Python code](https://github.com/opencdms/opencdms-process/pull/29) that calls R functions in the [RInstatClimatic R package](https://github.com/IDEMSInternational/cdms.products). 
The Python code converts a Pandas data frame to an R data frame using:
```
    with conversion.localconverter(default_converter + pandas2ri.converter):
        r_dataframe = conversion.py2rpy(python_dataframe)
```

An example Pandas data frame is shown below:

![image](https://user-images.githubusercontent.com/57253949/168238418-06fc6ca9-6650-41b7-9688-1cc4a231682c.png)

Note that the 'date' column does not specify the time zone.

If the time zone is not specified, then `py2rpy()` assumes that the date/time is in the system’s default timezone, and converts the time to UTC.

For example, if the date/time is specified without timezone (e.g. `1945-01-01`), and the system's timezone is GMT+1, then `py2rpy()` will convert `1945-01-01` to `1944-12-31T23:00:00Z`.

However, this is not the behaviour we want. **We want to tell rpy2 that the 'date' column contains UTC dates, not local timezone dates.**

The [Rpy2 documentation](https://rpy2.github.io/doc/v3.4.x/html/pandas.html) states:

> The timezone used for conversion is the system’s default timezone unless pandas2ri.default_timezone is specified… or unless the time zone is specified in the original time object. 

I looked in the source code and `pandas2ri.default_timezone` does not exist.
However, I did find a `default_timezone` in [robjects.vectors](rpy2/rpy2/rpy2/robjects/vectors.py). 
I tried the following:

```
    vectors.default_timezone = pytz.timezone("UTC")
    with conversion.localconverter(default_converter + pandas2ri.converter):
        r_params[key] = conversion.py2rpy(r_params[key])
```
However, `py2rpy()` still converted `1945-01-01` to `1944-12-31T23:00:00Z`.

**In summary, I believe the rpy2 documentation is incorrect. Please could the documentation be updated to explain how to specify that date/time columns contain UTC dates, not local timezone dates? Many thanks**

Further background can be found in this [PR](https://github.com/opencdms/opencdms-process/pull/29) and this [issue](https://github.com/opencdms/opencdms-process/issues/30).

**Comment by lloyddewit** [comment]
@lgautier , thank you for self-assigning this issue.

Update:

I worked around the problem by explicitly setting any 'naive' timestamps (e.g. no time zone info) in the Python data frame to 'aware' UTC timestamps:

```
    # If dates in data frame do not include timezone data, then set to UTC
    python_dataframe["date"] = to_datetime(python_dataframe["date"], utc=True)

    with conversion.localconverter(default_converter + pandas2ri.converter):
        r_dataframe = conversion.py2rpy(python_dataframe)
```

However, it would be much cleaner if I could use the Python data frame as provided, without changing it, and then instruct rpy2 to treat any 'naive' timestamps as UTC.
So if there's a chance that this issue could be resolved, then it would still be much appreciated.
Thanks :)

**Comment by lgautier** [comment]
Yes, the documentation was wrong about the location of `default_timezone`. It used to be in `pandas2ri` but was eventually moved to `vectors`. This should be fixed with the next build of the doc.

Otherwise it looks like an oversight in the conversion rules for pandas. `default_timezone` is used from R to pandas, but not the other way around.

**Comment by lgautier** [comment]
Timezone handling in Python and pandas is definitely a moving target. I am seeing that pandas is making API-breaking changes, and Python 3.9 is introducing timezone management in the stdlib (https://docs.python.org/3/library/zoneinfo.html). I'll need to catch with all those to think about how to best fix the issue.

**Comment by kenahoo** [comment]
I would say that the data shown in the spreadsheet most closely matches R's `Date` class, which has 1-day granularity and also has no notion of timezone: https://stat.ethz.ch/R-manual/R-devel/library/base/html/Dates.html.  Its guts are a simple integer count giving number of days after (or before, for negatives) 1970-01-01.

If there's an implicit exogenous timezone in this particular data, then it's probably better to convert it to a tz-aware datetime with both date & time specified, materializing whatever assumptions are appropriate for this data.

**Comment by lloyddewit** [comment]
@kenahoo Thank you for the response!

> If there's an implicit exogenous timezone in this particular data, then it's probably better to convert it to a tz-aware datetime with both date & time specified, materializing whatever assumptions are appropriate for this data.

Yes, that's what rpy2 currently does when it converts from pandas to R. The issue I have is that rpy2 always assumes that it needs to convert the time from the system's local timezone to UTC. For example, if the pandas date/time is specified without timezone (e.g. `1945-01-01`), and the system's timezone is GMT+1, then `py2rpy()` will convert pandas `1945-01-01` to R **`1944-12-31T23:00:00Z`**.

However, this is not the behaviour we want. **We want to tell rpy2 to assume that the pandas 'date' column contains UTC dates, not local timezone dates.**

**Comment by lgautier** [comment]
There is more than a mere documentation here it seems. I am happy to review a PR if anyone can get to this before I have the time to.

**Comment by lgautier** [comment]
It looks like pandas 2.0 is triggering failed unit test that will force addressing this (issue #1013).
